spec/models/manageiq/providers/lenovo/physical_infra_manager/refresher_spec.rb

```diff
@@ -49,4 +49,31 @@
   it 'will execute post_process_refresh_classes' do
     expect(refresher.post_process_refresh_classes).to eq([])
   end
+
+  it 'will perform a full refresh' do
+    # Perform the refresh twice to verify that a second run with existing data
+    # does not change anything
+    2.times do
+      EmsRefresh.refresh(ems)
+      ems.reload
+
+      assert_table_counts
+      assert_guest_table_contents
+    end
+  end
+
+  def assert_table_counts
+    expect(PhysicalServer.count).to eq(2)
+    expect(GuestDevice.count).to eq(5)
+  end
+
+  def assert_guest_table_contents
+    nic = GuestDevice.where(:device_name => "Broadcom 2-port 1GbE NIC Card for IBM").first
```

Comment:
Probably better going through a physical_server association since that name doesn't look like it'd be particularly unique